src/rust/ensogl/src/display/style.rs

```diff
@@ -0,0 +1,686 @@
+            (Data::Number(lhs),Data::Number(rhs)) => Data::Number(lhs+rhs),
+            // _ => Data::Invalid("Cannot multiply.".into())
+        }
+    }
+}
+
+impl Eq for Data {}
+
+
+
+// ============
+// === Path ===
+// ============
+
+/// Path is a set of strings which describes how the variable or style sheet is nested in the
+/// cascading style sheet map.
+#[derive(Clone,Debug)]
+#[allow(missing_docs)]
+pub struct Path {
+    pub segments : Vec<String>
+}
+
+impl Path {
+    /// Builds the path from the provided segment iterator. Please note that the internal path
+    /// representation is reversed, as the style sheet dependencies in the style sheet map tree are
+    /// also kept in a reversed order. Please use the `visualize` utility to inspect the internal
+    /// structure and learn more.
+    pub fn from_segments<T,I,Item>(t:T) -> Self
+        where T : IntoIterator<IntoIter=I,Item=Item>,
+              I : Iterator<Item=Item>,
+              I : DoubleEndedIterator,
+              Item : Into<String> {
+        Self::from_rev_segments(t.into_iter().rev())
+    }
+
+    /// Builds the path from reversed segment iterator. See `from_segments` to learn more.
+    pub fn from_rev_segments<T,Item>(t:T) -> Self
+        where T : IntoIterator<Item=Item>,
+              Item : Into<String> {
+        let segments = t.into_iter().map(|s|s.into()).collect();
+        Self {segments}
+    }
+}
+
+impl From<&str> for Path {
+    fn from(t:&str) -> Self {
+        Self::from_rev_segments(t.rsplit("."))
+    }
+}
+
+impl From<&Path> for Path {
+    fn from(t:&Path) -> Self {
+        t.clone()
+    }
+}
+
+macro_rules! gen_var_path_conversions {
+    ($($($num:tt)?),*) => {$(
+        impl<T:?Sized> From<&[&T$(;$num)?]> for Path
+        where for <'t> &'t T : Into<String> {
+            fn from(t:&[&T$(;$num)?]) -> Self {
+                Self::from_segments(t.into_iter().map(|s|*s))
+            }
+        }
+    )*};
+}
+
+gen_var_path_conversions!(1,2,3,4,5,6,7,8,9,10,);
+
+
+
+// ===========
+// === Var ===
+// ===========
+
+/// Data of a style variable. Variables are associated with a style path like 'panel.button.size'
+/// and are automatically bound to the most specific style sheet as soon as it gets defined. By
+/// most specific, we mean the one with the longest path. For example, the 'panel.button.size' var
+/// will be bound to one of 'panel.button.size', 'button.size', or 'size' if defined, in that order.
+#[derive(Debug)]
+pub struct Var {
+    /// Index of the var in the style var map.
+    pub index : Index<Var>,
+    /// Set of all `Sheet` indexes which are potential matches of this var. For example, for a var
+    /// 'panel.button.size', all of the following sheets will be included here: 'panel.button.size',
+    /// 'button.size', and 'size'.
+    pub matches : Vec<Index<Sheet>>,
+    /// Index of the most specific `Sheet` from `matches` which has a defined value if any.
+    pub binding : Option<Index<Sheet>>,
```

Comment:
Do we really want to those two fields to being public, if they have somewhat "constrained" relation? One could just set something to `binding` without caring about the `matches` content.

Reply:
Yeah, these (and other) fields should be private. Fixing.